include/Ark/Compiler/Lexer.inl

```diff
@@ -15,15 +13,14 @@ inline TokenType Lexer::guessType(const std::string& value) noexcept
         return TokenType::Operator;
     else if (isKeyword(value))
         return TokenType::Keyword;
-    else if (value[0] == '&' && value.size() > 1 && CHECK_FIRST_CHAR(value[1]))
+    else if (value.size() > 3 && value[0] == value[1] && value[1] == value[2] && value[2] == '.')
+        return TokenType::Spread;
+    else if (value[0] == '&' && value.size() > 1 && isIdentifier(value))
         return TokenType::Capture;
-    else if (value[0] == '.' && value.size() > 1 && CHECK_FIRST_CHAR(value[1]))
+    else if (value[0] == '.' && value.size() > 1 && isIdentifier(value))
         return TokenType::GetField;
-    // otherwise, identifier if it starts with [a-zA-Z_]
-    else if (CHECK_FIRST_CHAR(value[0]))
+    else if (isIdentifier(value))
         return TokenType::Identifier;
-    else if (value.size() > 3 && value[0] == value[1] && value[1] == value[2] && value[2] == '.')
-        return TokenType::Spread;
```

Comment:
without thise, macros can't work

Reply:
okay, fix coming

Reply:
but if you read entirely the file Spread token type is still return

Reply:
ow, ok i will back to previous version

Reply:
No I meant it wasn't visible on the diff, the current version is correct

Reply:
so, it's good no change needed